### PR TITLE
BFD-164: AB2D glue role

### PR DESCRIPTION
**Pull request by RickHawesUSDS**
**JIRA Ticket:**
BFD-164

**User Story or Bug Summary:**

Setup the BFD-Insights project for AB2D

**What Does This PR Do?**

This PR adds an AWS role for Glue jobs to access the S3 buckets of the project. 

**What Should Reviewers Watch For?**

Mistakes or missing pieces. 

**Submitters should complete the following questionnaire:**
N/A

### CODE REVIEW

File: insights/terraform/modules/bucket/main.tf [hunk 1]
```diff
@@ -47,6 +47,8 @@ resource "aws_s3_bucket_object" "top" {
 resource "aws_kms_key" "main" {
   description   = "CMK for the ${local.full_name} bucket"
   tags          = var.tags
+  key_usage     = "ENCRYPT_DECRYPT"
+  is_enabled    = true
   policy        = length(var.cross_accounts) > 0 ? data.aws_iam_policy_document.cmk_policy.json : null
```
Comment (RickHawesUSDS) [inline insights/terraform/modules/bucket/main.tf:52]: This change makes defaults explicit

File: insights/terraform/modules/bucket/outputs.tf [hunk 2]
```diff
@@ -9,3 +9,7 @@ output "id" {
 output "bucket_cmk" {
   value = aws_kms_key.main.key_id
 }
+
+output "bucket_cmk_arn" {
+  value = aws_kms_key.main.arn
+}
```
Comment (RickHawesUSDS) [inline insights/terraform/modules/bucket/outputs.tf:15]: Will need the `arn` later.

File: insights/terraform/modules/bucket/policies.tf [hunk 3]
```diff
@@ -187,11 +189,11 @@ resource "aws_s3_bucket_policy" "cross_account" {
...
                   "s3:GetBucketLocation",
-                  "s3:GetObject",
+                  "s3:GetObject*",
                   "s3:ListBucket",
                   "s3:ListBucketMultipartUploads",
-                  "s3:PutObject",
-                  "s3:PutObjectAcl"
+                  "s3:PutObject*",
+                  "s3:DeleteObject*"
               ],
```
Comment (RickHawesUSDS) [inline insights/terraform/modules/bucket/policies.tf:197]: Expanding the policy to be more comprehensive.

File: insights/terraform/projects/ab2d/main.tf [hunk 4]
```diff
@@ -46,148 +54,17 @@ module "database" {
   tags            = local.tags
 }
 
-## Tables for the project
```
Comment (RickHawesUSDS) [inline insights/terraform/projects/ab2d/main.tf:49]: Moving these tables to a new `tables.tf` as refactoring.

File: insights/terraform/modules/jobs/main.tf [hunk 5]
```diff
@@ -0,0 +1 @@
+# Glue jobs stuff goes here
```
Comment (RickHawesUSDS) [inline insights/terraform/modules/jobs/main.tf:1]: The Glue Jobs module will have a lot more in the future.

File: insights/terraform/modules/jobs/policies.tf [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  {
+    "Version": "2012-10-17",
+    "Statement": [
+      {
+        "Action": "sts:AssumeRole",
+        "Principal": {
+          "Service": "glue.amazonaws.com"
+        },
+        "Effect": "Allow",
+        "Sid": ""
```
Comment (njdister) [inline insights/terraform/modules/jobs/policies.tf:19]: Somewhat inconsequential, but the SID is included, yet blank.

File: insights/terraform/modules/jobs/policies.tf [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+data "aws_iam_policy" "glue_service" {
+  arn  = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
+}
+
+data "aws_iam_policy" "athena_service" {
+  arn  = "arn:aws:iam::aws:policy/AmazonAthenaFullAccess"
+}
+
+data "aws_iam_policy_document" "s3_access" {
```
Comment (njdister) [inline insights/terraform/modules/jobs/policies.tf:36]: Interesting way to generate policy documents!

File: insights/terraform/projects/ab2d/main.tf [hunk 8]
```diff
@@ -25,6 +25,14 @@ module "bucket" {
   ]
 }
 
+data "aws_s3_bucket" "moderate_bucket" {
+  bucket = "bfd-insights-moderate-577373831711"
```
Comment (njdister) [inline insights/terraform/projects/ab2d/main.tf:29]: Suggest replacing this with the caller identity data source

File: insights/terraform/modules/jobs/policies.tf (hunk 7 above)
Comment (RickHawesUSDS) [reply insights/terraform/modules/jobs/policies.tf:36]: Right. JSON and HEREDOC are harder to debug.

File: insights/terraform/projects/ab2d/main.tf (hunk 8 above)
Comment (RickHawesUSDS) [reply insights/terraform/projects/ab2d/main.tf]: Good point. Pushed a new change.

**Pull request merged**